core/server/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -52,7 +52,9 @@ public void close() {}
 
     @Override
     protected ClientRWLock createNewResource() {
-      return new ClientRWLock();
+      ClientRWLock lock = new ClientRWLock();
+      lock.addReference();
```

Comment:
My two cents,  not really critical, but just my own opinion for code cleanness:
I feel it is a little bit wired / confusing to increase the ref count in this subclass, without ref count dec in the same subclass. One alternative way is to keep the resource pool impl as is, but move `lock.addReference();` right after  `mLocks.put(blockId, blockLock);` in `getBlockLock(()`. In that case, I think it is still equivalent to the current PR, but having one (slight) advantage: having refcount inc only in `getBlockLock` ( you will not need refcount dec in that method any more) and having refcount dec only in `releaseBlockLockIfUnused`, which looks more aligned.